nansat/mappers/mapper_netcdf_cf.py

```diff
@@ -362,26 +362,18 @@ def _create_empty_with_nansat_spatial_reference_WKT(self, gdal_dataset, gdal_met
             and set the projection to the Nansat Spatial Reference WKT
             [NSR().wkt], using the first subdataset as source
         """
-        try:
-            xs = gdal_dataset.RasterXSize
-            ys = gdal_dataset.RasterYSize
-            gt = gdal_dataset.GetGeoTransform()
-        except Exception as e:
-            # issue warning to find out if this try-except clause is really needed...
-            warnings.warn('Main GDAL dataset does not contain geotransform or information ' \
-                    'about raster size: %s' %e.message)
-            fn = self._get_sub_filenames(gdal_dataset)
-            sub = gdal.Open(fn[0])
-            xs = sub.RasterXSize
-            ys = sub.RasterYSize
-            gt = sub.GetGeoTransform()
+        fn = self._get_sub_filenames(gdal_dataset)
```

Comment:
this is fewer lines and looks ok, but still the call to `_get_sub_filenames` could be avoided by first trying to read the attributes of `gdal_dataset` itself...